satellite/satellitedb/projectmembers.go

```diff
@@ -20,42 +20,73 @@ type projectMembers struct {
 	db      *dbx.DB
 }
 
-// GetByMemberID is a method for querying project member from the database by memberID.
-func (pm *projectMembers) GetByMemberID(ctx context.Context, memberID uuid.UUID) (_ []console.ProjectMember, err error) {
+func (pm *projectMembers) GetByProjectIDTotal(ctx context.Context, projectID uuid.UUID, cursor console.ProjectMembersCursor) (_ *console.ProjectMembersPage, err error) {
 	defer mon.Task()(&ctx)(&err)
-	projectMembersDbx, err := pm.methods.All_ProjectMember_By_MemberId(ctx, dbx.ProjectMember_MemberId(memberID[:]))
-	if err != nil {
-		return nil, err
-	}
 
-	return projectMembersFromDbxSlice(ctx, projectMembersDbx)
-}
+	search := "%" + strings.Replace(cursor.Search, " ", "%", -1) + "%"
 
-// GetByProjectID is a method for querying project members from the database by projectID, offset and limit.
-func (pm *projectMembers) GetByProjectID(ctx context.Context, projectID uuid.UUID, pagination console.Pagination) (_ []console.ProjectMember, err error) {
-	defer mon.Task()(&ctx)(&err)
-	if pagination.Limit < 0 || pagination.Offset < 0 {
-		return nil, errs.New("invalid pagination argument")
+	if cursor.Limit > 50 {
+		cursor.Limit = 50
 	}
 
-	var projectMembers []console.ProjectMember
-	searchSubQuery := "%" + strings.Replace(pagination.Search, " ", "%", -1) + "%"
-	//`+getOrder(pagination.Order)+`
+	if cursor.Page == 0 {
+		return nil, errs.New("page cannot be 0")
+	}
+
+	page := &console.ProjectMembersPage{
+		Search: cursor.Search,
+		Limit:  cursor.Limit,
+		Offset: uint64((cursor.Page - 1) * cursor.Limit),
+	}
 
+	countQuery := pm.db.Rebind(`
+		SELECT COUNT(*)
+		FROM project_members pm 
+		INNER JOIN users u ON pm.member_id = u.id
+			WHERE pm.project_id = ?
+			AND ( u.email LIKE ? OR 
+				  u.full_name LIKE ? OR
+				  u.short_name LIKE ? 
+			)`)
+
+	countRow := pm.db.QueryRowContext(ctx,
+		countQuery,
+		projectID[:],
+		search,
+		search,
+		search)
+
+	err = countRow.Scan(&page.TotalCount)
+	if err != nil {
+		return nil, err
+	}
+	if page.TotalCount == 0 {
+		return page, nil
+	}
+	if page.Offset > page.TotalCount - 1 {
+		return nil, errs.New("page is out of range")
+	}
 	// TODO: LIKE is case-sensitive postgres, however this should be case-insensitive and possibly allow typos
 	reboundQuery := pm.db.Rebind(`
 		SELECT pm.*
 			FROM project_members pm
 				INNER JOIN users u ON pm.member_id = u.id
 					WHERE pm.project_id = ?
-					AND ( u.email LIKE ? OR
+					AND ( u.email LIKE ? OR 
```

Comment:
Remove this change, it's just a space at the end of the line, before break line character.